src/main/java/ca/uhn/fhir/jpa/starter/AuthenticationInterceptor.java

```diff
@@ -44,7 +48,11 @@ public boolean incomingRequestPostProcessed(HttpServletRequest theRequest,
             
             // Here we test for a hardcoded username & password use to get access to the
             // hapi fhir JPA sever..
-            if (!username.equals("hapi") || !password.equals("hapi123")) {
+            String appConfigPath = Thread.currentThread().getContextClassLoader().getResource("login.properties").getPath();
+            Properties appProps = new Properties();
+            appProps.load(new FileInputStream(appConfigPath));
+            
+            if (!username.equals(appProps.getProperty("username")) || !password.equals(appProps.getProperty("password"))) {
```

Comment:
what if we simply added the one exclamation mark ie      if(!(username.equals(appProps.getProperty("username")) ||   password.equals(appProps.getProperty("password")))){

  }